Amplify/Categories/DataStore/Query/ModelKey.swift

```diff
@@ -32,24 +32,42 @@ import Foundation
 ///     post.content != nil
 /// })
 /// ```
-public protocol ModelKey: CodingKey, CaseIterable, QueryFieldOperation {}
+public protocol ModelKey: CodingKey, CaseIterable, QueryFieldOperation {
+    var modelName: String { get }
+}
+
+extension ModelKey {
+    public var modelName: String { "" }
```

Comment:
document the behavior. Why is it empty by default? (i.e. explain the backward compatibility)